egui_demo_lib/src/apps/demo/paint_bezier.rs

```diff
@@ -0,0 +1,233 @@
+            let radius = aux_stroke.width * 3.0;
+            let circle = CircleShape {
+                center,
+                radius,
+                fill: aux_stroke.color,
+                stroke: aux_stroke.clone(),
+            };
+            
+            shapes.push(circle.into());
+        }
+        
+        shapes
+    }
+
+    pub fn ui_content(&mut self, ui: &mut Ui) -> egui::Response {
+        let (mut response, painter) =
+            ui.allocate_painter(ui.available_size_before_wrap(), Sense::click());
+
+        painter.ctx().set_tolerance(self.tolerance);
+        let to_screen = emath::RectTransform::from_to(
+            Rect::from_min_size(Pos2::ZERO, response.rect.square_proportions()),
+            response.rect,
+        );
+        let from_screen = to_screen.inverse();
+
+        if response.clicked() {
+            if let Some(pointer_pos) = response.interact_pointer_pos() {
+                let canvas_pos = from_screen * pointer_pos;
+                self.points.push(canvas_pos);
+                if self.points.len() >= self.bezier {
+                    self.backup_points = self.points.clone();
+                    let points = self.points.drain(..).collect::<Vec<_>>();
+                    match points.len(){
+                        3 => {
+                            let quadratic = QuadraticBezierShape::from_points_stroke(
+                                    points, 
+                                    self.closed,
+                                    self.fill.clone(),
+                                    self.stroke.clone()
+                                );
+                            self.q_shapes.push(quadratic);
+                        },
+                        4 => {
+                            let cubic = CubicBezierShape::from_points_stroke(
+                                points, 
+                                self.closed,
+                                self.fill.clone(),
+                                self.stroke.clone()
+                            );
+                            self.c_shapes.push(cubic);
+                        },
+                        _ => {
+                            todo!();
+                        }
+                    }
+                }
+                
+                response.mark_changed();
+            } 
+        }
+        let mut shapes = Vec::new();
+        for shape in self.q_shapes.iter(){
+            shapes.push(shape.to_screen(&to_screen).into());
+            if self.show_bounding_box{
+                shapes.push(self.build_bounding_box(shape.bounding_rect(),&to_screen));
+            }
+        }
+        for shape in self.c_shapes.iter(){
+            shapes.push(shape.to_screen(&to_screen).into());
+            if self.show_bounding_box{
+                shapes.push(self.build_bounding_box(shape.bounding_rect(),&to_screen));
+            }
+        }
+        painter.extend(shapes);
+
+        if self.points.len()>0{
```

Comment:
Hot tip: configure your editor to run `cargo fmt` on each save. It is a life-changer :)